edb/graphql/types.py

```diff
@@ -1248,6 +1248,13 @@ def _define_types(self) -> None:
                 # The aliased types actually only reflect as an object
                 # type, but the rest of the processing is identical to
                 # interfaces.
+                if '~' in gql_name:
+                    # These are internal aliases, but they define the
+                    # actual shapes we need to reflect.
+                    gql_name = '_edb' + gql_name.replace('|', '_') \
```

Comment:
Ugh, this is unfortunate.  I have a branch where nested alias naming is fixed properly.  If we want to fix this here now, ideally we'd need to synchronize the mangling schema.